File: Stack.cpp
#include "Stack.h"


typedef int T;



Stack::Stack() {
		Container = new T[capasity];
	}
	
Stack::Stack(std::initializer_list<T> list) {
		this->capasity = list.size();

		Container = new T[capasity];

		for (auto item : list)
			Add(item);
	}

bool Stack::isEmpty() {
		return Top == -1;
	}

void Stack::Add(T data) {
		if (Top + 1 == capasity) { std::cout << "Stack is full" << std::endl; return; }
		Top++;
		this->Container[Top] = data;
	}

void Stack::Delete() {
		if (isEmpty())std::cout << "Stack is emty" << std::endl;
		Top--;
	}

void Stack::Dispaly(std::ostream& out) {

		if (isEmpty()) {
			std::cout << "List is empty can't display anything" << std::endl;

			return;
		}

		for (int i = 0; i <= Top; i++) {
			out << this->Container[i] << "\t";
		}
	}

std::ostream& operator <<(std::ostream& out, Stack& S1) {

	S1.Dispaly(out);
	return out;

}

File: Stack.cs
using System;

namespace MainNS{

    class StaticStack<T>{
        int size = 0;
        const int capacity = 10;
        T[] container = new T[capacity];
        public StaticStack(){
            
        }

        public StaticStack(in StaticStack<T> Zoro){
            foreach(T item in Zoro.container)
                Add(item);
        }

        public void Add(T data){
            if(!IsFull()){
                this.container[size] = data;
                size++; 
            }else{
                Console.WriteLine("Stack is full!!!");
            }

        }
        public void Delete(){
            if(!IsEmpty())
                size--;
            else 
                Console.WriteLine("Stack is already empty");
        }
        public void Display(){
            for(int i = 0; i < size;i++)
                Console.Write(this.container[i] + "\t");
            Console.WriteLine();
        }
        public bool IsEmpty(){
            return this.size == 0;
        }
        public bool IsFull(){
            return this.size == capacity;
        }

    }

}